public/dorequest.php

```diff
@@ -388,6 +448,56 @@ function DoRequestError(string $error, ?int $status = 200, ?string $code = null)
         }
         break;
 
+    case "syncachievements":
+        $achievementIdsInput = request()->input('a', '');
+        $hardcore = (bool) request()->input('h', 0);
+
+        $foundTargetUser = null;
+        if (!$delegateTo) {
+            return DoRequestError("You must specify a target user.", 400);
+        }
+
+        $foundTargetUser = User::firstWhere('User', $delegateTo);
+        if (!$foundTargetUser) {
+            return DoRequestError("The target user couldn't be found.", 400);
+        }
+
+        $achievementIdsArray = explode(',', $achievementIdsInput);
+        $idsToAttemptAward = array_filter($achievementIdsArray, function ($id) {
+            return filter_var($id, FILTER_VALIDATE_INT) !== false;
+        });
+
+        if (count($idsToAttemptAward) > 800) {
```

Comment:
I would hope we don't try to award more than a couple dozen achievements at a time. The initial unlock of all previously earned achievements might have to be batched.

Reply:
> The initial unlock of all previously earned achievements might have to be batched.

Can you explain this another way? There will almost certainly be a case where an initial "sync" contains 100+ achievements, and I should ensure how we expect this to be handled is well-documented.

Reply:
> Can you explain this another way?

For normal use, I would only expect a handful of achievements to unlocked at a time - kill boss, kill boss damageless, kill boss using weapon X, complete level in under Y seconds, etc. Having 800 unlocks at the same time would be very spammy.

I understand that this is meant primarily for backfilling achievements earned before integrating with RetroAchievements, but it could also be used to batch several new unlocks together in a single server call - there could even be desire to do that for non-standalone emulators.

800 unlocks in a single request (even for backfilling) seems excessive. Does that imply that the standalone sets will have more than 800 achievements? That just seems unmanageable. If there needs to be an upper limit, it should be something more in line with the expected use cases.

If 1000 unlocks is a valid backfill use case, then the client should batch that into smaller chunks to fit whatever we determine to be a reasonable constraint. 800 doesn't feel like a reasonable constraint - you might as well leave it unbounded.

Reply:
The upper bound has been removed. The new logic makes a query to `player_achievements` to efficiently determine which achievements an unlock should be attempted for.